tests/modes.spec.ts

```diff
@@ -261,6 +290,21 @@ test("the viewport changes only once when opening a form", async () => {
   expect(new URL(page.url()).hash).toEqual(customViewport);
 });
 
+test("switching between modes saves, doesn't save anything if no actions taken", async () => {
+  await page.getByRole("button", { name: "New polygon (snapped)" }).click();
+  await clickMap(page, 500, 500);
+  await clickMap(page, 400, 500);
+  await clickMap(page, 400, 600);
+
+  await page.getByRole("button", { name: "New route" }).click();
+  await expect(
+    page.getByRole("button", { name: "1) Untitled area" })
+  ).toBeVisible();
+
+  await page.getByRole("button", { name: "New polygon (snapped)" }).click();
+  await expect(page.getByText("1 objects")).toBeVisible();
```

Comment:
More resilient, too, in case there is a delay in creation of the area/route object

Reply:
>  in case there is a delay in creation of the area/route object

Good point. Playwright auto-await is smart about polling for the existence of locators, but not sure how it's possible for it to confirm non-existence of something other than just sleeping